core/unit_test/TestHalfOperators.hpp

```diff
@@ -910,12 +908,13 @@ struct Functor_TestHalfOperators {
 
 template <class half_type>
 void __test_half_operators(half_type h_lhs, half_type h_rhs) {
-  double epsilon = FLT_EPSILON;
-
-  if (std::is_same<half_type, Kokkos::Experimental::half_t>::value)
-    epsilon = FP16_EPSILON;
-  if (std::is_same<half_type, Kokkos::Experimental::bhalf_t>::value)
-    epsilon = BF16_EPSILON;
+  double epsilon = Kokkos::Experimental::epsilon<half_type>::value;
+#if defined(KOKKOS_BHALF_T_IS_FLOAT) && KOKKOS_BHALF_T_IS_FLOAT
+  double constexpr epsilon_bias = 10e3;
+#else
+  double constexpr epsilon_bias = 1;
+#endif
```

Comment:
Why do you need a bias?

Reply:
Good question. Before moving these constants to numeric traits, when bhalf_t or half_t were aliases to float, the tests were erroneously using FP16_EPSILON and BF16_EPSILON from:
```
 double epsilon = FLT_EPSILON;

  if (std::is_same<half_type, Kokkos::Experimental::half_t>::value)
    epsilon = FP16_EPSILON;
  if (std::is_same<half_type, Kokkos::Experimental::bhalf_t>::value)
    epsilon = BF16_EPSILON;
```

After further triaging, I found that the only test failing without the bias was PASS_BY_REFF. The way the PASS_BY_REF test was written resulted in the compiler optimizing out the cast from float -> double. See the latest commit for a new way of writing that test.